fameal/ape-tax
# Feature/Ape Haussmann's renovation

## What it does ✨
> Beginning in 1854, in the centre of the city, Haussmann's renovation tore down hundreds of old buildings and cut eighty kilometres of new avenues. It included the demolition of medieval neighbourhoods that were deemed overcrowded and unhealthy by officials at the time; the building of wide avenues; new parks and squares; the annexation of the suburbs surrounding Paris; and the construction of new sewers, fountains and aqueducts.

This PR is a large refactoring and rewriting of the ape.tax codebase from Vue to React. Several elements motivated this choice: 
- The large community of dev behind React, and the vast number of packages, components and plugins for the Web3 development. 🦍
- The old codebase didn't follow the Vue standards (written in Python style, which is not standard). 🚨
- Some nice feature (see bellow) would have taken more time to implement in Vue than rewriting the whole dAPP and theses features in React. ⏰

The process was quite simple : the design was respected and reproduced, but the whole code has been rewritten with NextJs + Tailwindcss. It will be easy to deploy this version with Vercel, but a few environment variables will need to be added
```
ALCHEMY_KEY -> ALCHEMY_KEY
SECRET -> "ArrowUp,ArrowUp,ArrowDown,ArrowDown,ArrowLeft,ArrowRight,ArrowLeft,ArrowRight,KeyB,KeyA"
WEBSITE_URI -> "https://ape.tax/"
```

The following features/improvements were made:
- Adding a support for `WalletConnect` 👛
- Reworking the 'Network' select to change the network for the user at the same time ♻️
- Removing the drizzle usage to migrate with a plain Ethers.js usage 📦
- Removing the credits for a more generic one for anon reason 👏
- Support of mobile version  📱
- Removing the reload between pages for the smoothest experience ⏳
- Support for mainnet-fork to easy dev 🧪

## How to test ✅
- Install the app with `yarn`.
- Launch the app with `yarn run dev`.
- You should be able to see the app. You should be able to connect with Metamask or walletConnect.
- You should be able to see the list of vaults. You should be able to click a vault and get the details about the vault, the strategies and your position.
- You should be able to approve, withdraw, deposit for any vault (if you have funds), and the vault status should be updated after any action.
- You should be able to change the network used with the top right select.

## Note ℹ️
Don't get me wrong : the previous version was working very nicely. This is just an improvement, but this is a big change. Only to merge if the maintainers think it's worth it!

As the Vercel preview may fail (deployment from Vue to React, [here is a preview](https://ape.major.farm/))
